`numpy.count_nonzero` returns `np.int64` instead of `int` with NumPy >= 2.3

This causes the `test_count_nonzero` tests to fail. They test for the return value to be of type `int`.

<details>
<summary>Output from pytest</summary>

```python
=================================== FAILURES ===================================
_________________________ test_count_nonzero[numpoly] __________________________

func_interface = <module 'numpoly' from '/builddir/build/BUILD/python-numpoly-1.3.8-build/BUILDROOT/usr/lib64/python3.14/site-packages/numpoly/__init__.py'>

    def test_count_nonzero(func_interface):
        """Tests for numpoly.count_nonzero."""
        poly1 = polynomial([[0, Y], [X, 1]])
        poly2 = polynomial([[0, Y, X, 0, 0], [3, 0, 0, 2, 19]])
>       assert_equal(func_interface.count_nonzero(poly1), 3, type_=int)

test/test_array_function.py:360: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

results = np.int64(3), reference = 3, c_contiguous = None, f_contiguous = None
type_ = <class 'int'>

    def assert_equal(results, reference, c_contiguous=None,
                     f_contiguous=None, type_=None):
        """
        Assert that a return value for a function is the same as reference.
    
        Checks types, values, datatype, shape, C- and F-contiguous-ness.
    
        Args:
            results (numpy.ndarray, numpoly.ndpoly):
                The results to check are correct.
            reference (numpy.ndarray, numpoly.ndpoly):
                The reference the results is checked against. Input will be cast to
                the same type as `results`.
            c_contiguous (Optional[bool]):
                Check if `results` has correct `C_CONTIGUOUS` flag. Checked against
                `reference` if not provided.
            f_contiguous (Optional[bool]):
                Check if `results` has correct `F_CONTIGUOUS` flag. Checked against
                `reference` if not provided.
            type_ (Optional[type]):
                Check if `results` is correct type using `isinstance`. If not
                provided, results are checked to be legal numpy or numpoly type.
    
        """
        if type_ is None:
            assert isinstance(
                results, (bool, numpy.bool_, numpy.number, numpy.ndarray)), (
                    f"unrecognized results type: {results}")
        else:
>           assert isinstance(results, type_), (
                f"invalid results type: {results} != {type_}")
E           AssertionError: invalid results type: 3 != <class 'int'>
E           assert False
E            +  where False = isinstance(np.int64(3), <class 'int'>)

test/test_array_function.py:40: AssertionError
__________________________ test_count_nonzero[numpy] ___________________________

func_interface = <module 'numpy' from '/usr/lib64/python3.14/site-packages/numpy/__init__.py'>

    def test_count_nonzero(func_interface):
        """Tests for numpoly.count_nonzero."""
        poly1 = polynomial([[0, Y], [X, 1]])
        poly2 = polynomial([[0, Y, X, 0, 0], [3, 0, 0, 2, 19]])
>       assert_equal(func_interface.count_nonzero(poly1), 3, type_=int)

test/test_array_function.py:360: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

results = np.int64(3), reference = 3, c_contiguous = None, f_contiguous = None
type_ = <class 'int'>

    def assert_equal(results, reference, c_contiguous=None,
                     f_contiguous=None, type_=None):
        """
        Assert that a return value for a function is the same as reference.
    
        Checks types, values, datatype, shape, C- and F-contiguous-ness.
    
        Args:
            results (numpy.ndarray, numpoly.ndpoly):
                The results to check are correct.
            reference (numpy.ndarray, numpoly.ndpoly):
                The reference the results is checked against. Input will be cast to
                the same type as `results`.
            c_contiguous (Optional[bool]):
                Check if `results` has correct `C_CONTIGUOUS` flag. Checked against
                `reference` if not provided.
            f_contiguous (Optional[bool]):
                Check if `results` has correct `F_CONTIGUOUS` flag. Checked against
                `reference` if not provided.
            type_ (Optional[type]):
                Check if `results` is correct type using `isinstance`. If not
                provided, results are checked to be legal numpy or numpoly type.
    
        """
        if type_ is None:
            assert isinstance(
                results, (bool, numpy.bool_, numpy.number, numpy.ndarray)), (
                    f"unrecognized results type: {results}")
        else:
>           assert isinstance(results, type_), (
                f"invalid results type: {results} != {type_}")
E           AssertionError: invalid results type: 3 != <class 'int'>
E           assert False
E            +  where False = isinstance(np.int64(3), <class 'int'>)

test/test_array_function.py:40: AssertionError
=============================== warnings summary ===============================
test/test_array_function.py::test_isfinite[numpoly]
test/test_array_function.py::test_isfinite[numpy]
  /builddir/build/BUILD/python-numpoly-1.3.8-build/numpoly-1.3.8/test/test_array_function.py:664: RuntimeWarning: invalid value encountered in log
    poly = numpoly.polynomial([numpy.log(-1.), X, numpy.log(0)])

test/test_array_function.py::test_isfinite[numpoly]
test/test_array_function.py::test_isfinite[numpy]
  /builddir/build/BUILD/python-numpoly-1.3.8-build/numpoly-1.3.8/test/test_array_function.py:664: RuntimeWarning: divide by zero encountered in log
    poly = numpoly.polynomial([numpy.log(-1.), X, numpy.log(0)])

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
=========================== short test summary info ============================
FAILED test/test_array_function.py::test_count_nonzero[numpoly] - AssertionEr...
FAILED test/test_array_function.py::test_count_nonzero[numpy] - AssertionErro...
================== 2 failed, 219 passed, 4 warnings in 9.82s ===================
```

</details>

A simple reproducer with Python 3.11.13:

```python
>>> import numpy as np
>>> import numpoly
>>>
>>> X, Y = numpoly.variable(2)
>>> poly1 = numpoly.polynomial([[0, Y], [X, 1]])
>>> numpoly.count_nonzero(poly1)
np.int64(3)
```

With Python 3.10.18 the output is `3`.

I applied below as a quick fix while testing. I'd be happy to submit a proper PR with a less ad hoc approach. While looking into this I came across https://github.com/numpy/numpy/issues/17562 which might give some insight.

```diff
@@ -357,10 +358,10 @@ def test_count_nonzero(func_interface):
     """Tests for numpoly.count_nonzero."""
     poly1 = polynomial([[0, Y], [X, 1]])
     poly2 = polynomial([[0, Y, X, 0, 0], [3, 0, 0, 2, 19]])
-    assert_equal(func_interface.count_nonzero(poly1), 3, type_=int)
+    assert_equal(func_interface.count_nonzero(poly1), 3, type_=(numpy.intp if sys.version_info >= (3, 11) else int))
     assert_equal(func_interface.count_nonzero(poly1, axis=0), [1, 2])
     assert_equal(func_interface.count_nonzero(poly2, axis=0), [1, 1, 1, 1, 1])
-    assert_equal(func_interface.count_nonzero(X), 1, type_=int)
+    assert_equal(func_interface.count_nonzero(X), 1, type_=(numpy.intp if sys.version_info >= (3, 11) else int))
 
 
 def test_cumsum(interface):
```

--- a/test/test_array_function.py
+++ b/test/test_array_function.py
@@ -4,6 +4,7 @@ from pytest import raises
 import numpy
 from numpoly import polynomial
 import numpoly
+from packaging.version import Version
 
 X, Y = numpoly.variable(2)
 
@@ -357,10 +358,16 @@ def test_count_nonzero(func_interface):
     """Tests for numpoly.count_nonzero."""
     poly1 = polynomial([[0, Y], [X, 1]])
     poly2 = polynomial([[0, Y, X, 0, 0], [3, 0, 0, 2, 19]])
-    assert_equal(func_interface.count_nonzero(poly1), 3, type_=int)
     assert_equal(func_interface.count_nonzero(poly1, axis=0), [1, 2])
     assert_equal(func_interface.count_nonzero(poly2, axis=0), [1, 1, 1, 1, 1])
-    assert_equal(func_interface.count_nonzero(X), 1, type_=int)
+    # As of NumPy 2.3.0 `count_nonzero` returns a scalar instead of int
+    # https://github.com/jonathf/numpoly/issues/126
+    if Version(numpy.__version__) >= Version("2.3.0"):
+        assert_equal(func_interface.count_nonzero(poly1), 3, type_=numpy.intp)
+        assert_equal(func_interface.count_nonzero(X), 1, type_=numpy.intp)
+    else:
+        assert_equal(func_interface.count_nonzero(poly1), 3, type_=int)
+        assert_equal(func_interface.count_nonzero(X), 1, type_=int)
 
 
 def test_cumsum(interface):

